apps/frontend/src/libs/components/date-input/date-input.tsx

```diff
@@ -0,0 +1,109 @@
+import { Calendar } from "react-calendar";
+import {
+	type Control,
+	type FieldPath,
+	type FieldValues,
+} from "react-hook-form";
+
+import { Icon, Popover } from "~/libs/components/components.js";
+import { formatDate } from "~/libs/helpers/helpers.js";
+import {
+	useCallback,
+	useFormController,
+	usePopover,
+} from "~/libs/hooks/hooks.js";
+
+import { addDays, getDifferenceInDays } from "./libs/helpers/helpers.js";
+import { type DateValue } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties<T extends FieldValues> = {
+	control: Control<T, null>;
+	maxRange?: number;
+	name: FieldPath<T>;
+};
+
+const DateInput = <T extends FieldValues>({
+	control,
+	maxRange,
+	name,
+}: Properties<T>): JSX.Element => {
+	const { field } = useFormController({
+		control,
+		name,
+	});
+
+	const { isOpened, onClose, onOpen } = usePopover();
+
+	const getSelectedDateRange = (): string => {
```

Comment:
What if we don't have one of the values for example we selected only start date? is this case somehow prevented or handled?

Reply:
This is prevented by library. We cannot select only start date. However, there is an issue when we don't pass default values, or pass only one to the useAppForm hook. I'll try to do something about it.